### PR TITLE
backupccl: update restore/with-pause test

### LINKED ISSUE #94093: backupccl: in restore/with-pause roachtest, pause at 25/50/75% progress, not every 15 mins
We've seen this test flake on several branches due to the hard corded 15 minutes pause request interval. To avoid these flakes, we should pause the job at fixed progress intervals instead: at 25%, 50%, and 75% progress. 

Jira issue: CRDB-22690

### PR DESCRIPTION
**Pull request by msbutler**
This patch replaces the restore2TB/nodes=10/with-pause test with the new restore/pause/tpce/80GB/aws/nodes=4/cpus=8 test. The test now uses the new 80GB tpce fixture, and always pauses the restore job after around 25%, 50%, 75% completion.

This patch also increases the portability of the new restore roachtest framework, allowing the new test to use its facilities. 

Going forward, this test will make it easier to benchmark and test future changes to restore checkpointing.

Fixes #94093
Informs #87843

Release note: None

### CODE REVIEW

**Comment by msbutler** [comment]
Pre frontier checkpointing metrics: this 80 GB Restore with three pauses + ~3 minutes of sleep  takes around 40 minutes, a throughput around 18 mb/s/node.

File: pkg/cmd/roachtest/tests/restore.go [hunk 1]
```diff
@@ -953,34 +945,94 @@ type restoreSpecs struct {
...
-		return errors.Wrapf(err, "full test specs: %s", sp.computeName(true))
+// run executes the restore, where target injects a restore target into the restore command.
+// Examples:
+// - "DATABASE tpce" will execute a database restore on the tpce cluster.
+// - "" will execute a cluster restore.
+func (sp *restoreSpecs) run(ctx context.Context, target string) error {
+	return sp.c.RunE(ctx, sp.c.Node(1), sp.restoreCmd(target, ""))
+}
+
+func (sp *restoreSpecs) runDetached(ctx context.Context, target string) (jobspb.JobID, error) {
```
Comment (lidorcarmel) [inline pkg/cmd/roachtest/tests/restore.go:988]: I think you can now delete the old runRestoreDetached?

**Review by lidorcarmel — APPROVED** [review]
lgtm.

File: pkg/cmd/roachtest/tests/restore.go (hunk 1 above)
Comment (msbutler) [reply pkg/cmd/roachtest/tests/restore.go:988]: yup, in a follow up PR, i plan delete all the old infra and bank tests :D

File: pkg/cmd/roachtest/tests/restore.go [hunk 2]
```diff
@@ -646,29 +656,21 @@ func registerRestore(r registry.Registry) {
...
-			require.NotEqual(t, 3, maxPauses, "the job should have paused at least once")
+			require.NotEqual(t, 0, pauseIndex, "the job should have paused at least once")
 		},
-	})
 
-	durationGauge := r.PromFactory().NewGaugeVec(prometheus.GaugeOpts{Namespace: registry.
-		PrometheusNameSpace, Subsystem: "restore", Name: "duration"}, []string{"test_name"})
+		// TODO(msbutler): to test the correctness of checkpointing, we should
+		// restore the same fixture without pausing it and fingerprint both restored
+		// databases.
```
[the remark this answers is not in the document]
Comment (msbutler) [reply pkg/cmd/roachtest/tests/restore.go:672]: yeah, i plan to do that once this fingerprint bug is addressed  https://github.com/cockroachdb/cockroach/pull/97916

**Comment by msbutler** [comment]
TFTR!

bors r=lidorcarmel

**Pull request merged**